src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5701,6 +5701,60 @@ CancellationCode StorageReplicatedMergeTree::killMutation(const String & mutatio
     return CancellationCode::CancelSent;
 }
 
+void StorageReplicatedMergeTree::removePartsFromFilesystem(const DataPartsVector & parts)
+{
+    const auto settings = getSettings();
+    if (settings->max_part_removal_threads > 1 && parts.size() > settings->concurrent_part_removal_threshold)
+    {
+        /// Parallel parts removal.
+
+        size_t num_threads = std::min(size_t(settings->max_part_removal_threads), parts.size());
+        ThreadPool pool(num_threads);
```

Comment:
Not sure if this is optimal to create pool, maybe better to use threads from global pool ?

Reply:
This is the thread from the global pool, see https://github.com/ClickHouse/ClickHouse/blob/8dbf721e5289709f2457caf584c54fe80a1f0434/src/Common/ThreadPool.h#L229